src/daemon/normalize.js

```diff
@@ -0,0 +1,12 @@
+const punycode = require('punycode') // eslint-disable-line node/no-deprecated-api
+
+module.exports = function normalize(id) {
+  return id
+    ? punycode
+        .encode(id)
+        .toLowerCase()
+        .replace(/[^a-z0-9-.]/g, '-')
+        .replace(/-+/g, '-')
```

Comment:
Because so many characters are replaced by hyphen, any run of multiple hyphens is collapsed to a single hyphen.